pkg/expr/mathexp/type_series.go

```diff
@@ -94,31 +48,35 @@ FIELDS:
 			}
 			timeSlice = append(timeSlice, *val)
 		}
-		nF := data.NewField(frame.Fields[timeIdx].Name, nil, timeSlice) // (labels are not used on time field)
-		nF.Config = frame.Fields[timeIdx].Config
-		frame.Fields[timeIdx] = nF
+		nF := data.NewField(timeField.Name, nil, timeSlice) // (labels are not used on time field)
+		nF.Config = timeField.Config
+		timeField = nF
+	default:
+		return s, fmt.Errorf("invalid type %s for time field", timeField.Type())
 	}
 
-	if !valueNullable { // make value nullable if it is not in the input
-		floatSlice := make([]*float64, 0, frame.Fields[valueIdx].Len())
-		for rowIdx := 0; rowIdx < frame.Fields[valueIdx].Len(); rowIdx++ {
-			val, ok := frame.At(valueIdx, rowIdx).(float64)
-			if !ok {
-				return s, fmt.Errorf("unexpected time type, expected float64 but got %T", val)
+	valueField := frame.Fields[valueIdx]
+	// if value field is not *float64, try to convert to it
+	if valueField.Type() != data.FieldTypeNullableFloat64 {
+		floatSlice := make([]*float64, 0, valueField.Len())
+		for rowIdx := 0; rowIdx < valueField.Len(); rowIdx++ {
+			val, err := valueField.NullableFloatAt(rowIdx)
+			if err != nil {
+				return s, fmt.Errorf("unexpected value type, expected float64 but got %T", valueField.Type().String())
 			}
-			floatSlice = append(floatSlice, &val)
+			floatSlice = append(floatSlice, val)
 		}
-		nF := data.NewField(frame.Fields[valueIdx].Name, frame.Fields[valueIdx].Labels, floatSlice)
-		nF.Config = frame.Fields[valueIdx].Config
-		frame.Fields[valueIdx] = nF
+		nF := data.NewField(valueField.Name, valueField.Labels, floatSlice)
+		nF.Config = valueField.Config
+		valueField = nF
 	}
 
 	fields := make([]*data.Field, 2)
-	fields[seriesTypeTimeIdx] = frame.Fields[timeIdx]
-	fields[seriesTypeValIdx] = frame.Fields[valueIdx]
+	fields[seriesTypeTimeIdx] = timeField
+	fields[seriesTypeValIdx] = valueField
 
-	frame.Fields = fields
-	s.Frame = frame
+	s.Frame = data.NewFrame(frame.Name, fields...)
```

Comment:
Does this work?

```suggestion
	s.Frame = data.NewFrame(frame.Name, timeField, valueField)
```

Reply:
SSE has been changed to only produce dataplane data, and NewSeries ensures that.

So if we are not doing this now, then it still needs to be dataplane data - so should add the appropate type and typeversion properties on meta.